### ISSUE TITLE
CarbonPeriod with Addition

### ISSUE THREAD

**Issue by sebastienjean76** [briannesbitt/Carbon#1337]
I searched everywhere but I did not find the solution.
I wish to have a range of date. For this range, I would like to add a number of weeks.

For example today, we are on May 30, 2018, I would like to have something like this with end date on July 13, 2018:

2018-05-30
2018-05-31
2018-06-01
2018-06-02
2018-06-03

2018-06-11
2018-06-12
2018-06-13
2018-06-14
2018-06-15
2018-06-16
2018-06-17

2018-06-25
2018-06-26
2018-06-27
2018-06-28
2018-06-29
2018-06-30
2018-07-01

2018-07-09
2018-07-10
2018-07-11
2018-07-12
2018-07-13

How can I do ?

Thank you :)

**Comment by imbrish** [comment]
Assuming that you want your range fixed and you want to skip every other week:
```php
$period = new CarbonPeriod('2018-05-30', '2018-07-13');

$startOfFirstWeek = Carbon::parse('2018-05-30')->startOfWeek();

$period->addFilter(function ($date) use ($startOfFirstWeek) {
    return $date->startOfWeek()->diffInWeeks($startOfFirstWeek) % 2 == 0;
});
```

**Comment by imbrish** [comment]
@kylekatarnls hmmm `$period->skip(7)`? :)

**Comment by sebastienjean76** [comment]
Yes i want implement it. 
Thanks @imbrish and @kylekatarnls

**Comment by kylekatarnls** [comment]
I recommend you to checkout the version-1.29, because if we do a version 1.30 it will start from the actual version-1.29 branch.

**Comment by imbrish** [comment]
@kylekatarnls I think @sebastienjean76 meant using solution of this issue in his project?

I could implement `skip` when I get some more free time. Actually I think more about something like `shift`, so it could accept both negative and positive integers, but also a callback to directly modify the current date (it requires some more thinking for the least).

**Comment by kylekatarnls** [comment]
@sebastienjean76 I close this issue for now, do not hesitate to submit a pull-request for the `skip` method.

**Comment by kylekatarnls** [comment]
The `->skip()` method will be available in 1.30+ and 2.0+: #1345
